packages/web-frontend/src/utils/periodGranularities.js

```diff
@@ -122,8 +122,11 @@ export function roundStartEndDates(granularity, startDate = moment(), endDate =
 
 export const formatMomentAsString = (date, granularity, format) =>
   granularity === 'week' || granularity === 'one_week_at_a_time'
-    ? date.startOf('W').format(format)
-    : date.format(format);
+    ? date
+        .clone()
+        .startOf('W')
+        .format(format)
+    : date.clone().format(format);
```

Comment:
Hurray for unit tests!

Reply:
Oh man it would be so hard to catch this if not for unit tests... Mutability is one of the key points of moment.js criticism

Reply:
Yeah I should have checked that `.startOf` didn't mutate the original, but I just assumed it wouldn't 😬. Three cheers for https://github.com/beyondessential/tupaia-backlog/issues/1303